pmlc/dialect/pxa/transforms/buffer_placement.cc

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 Intel Corporation
+
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/Interfaces/SideEffectInterfaces.h"
+#include "mlir/Support/DebugStringHelper.h"
+
+#include "pmlc/dialect/pxa/analysis/uses.h"
+#include "pmlc/dialect/pxa/transforms/pass_detail.h"
+#include "pmlc/util/logging.h"
+
+using namespace mlir; // NOLINT[build/namespaces]
+
+namespace pmlc::dialect::pxa {
+
+struct BufferPlacementPass : public BufferPlacementBase<BufferPlacementPass> {
+  void runOnFunction() final {
+    auto fn = getFunction();
+    fn.walk([&](AllocOp alloc) {
+      IVLOG(3, "alloc: " << debugString(*alloc));
+
+      Operation *lastOp = alloc;
+      Block *allocBlock = alloc.getOperation()->getBlock();
+      for (auto &itUse : getIndirectUses(alloc)) {
+        Operation *use = itUse.getOwner();
+        IVLOG(3, "  use: " << debugString(*use));
+
+        while (use->getBlock() != allocBlock) {
```

Comment:
You can replace this loop with:
use = allocBlock->findAncestorOpInBlock(use);
assert(use && "no common ancestor)

Reply:
Done.